components/src/dynamo/sglang/request_handlers/image_diffusion/image_diffusion_handler.py

```
@@ -230,6 +231,44 @@ async def _upload_to_fs(
+
+        buf = io.BytesIO()
+        img.save(buf, format="PNG")
+        return buf.getvalue()
+
+    @staticmethod
+    def _ndarray_to_png(arr) -> bytes:
+        """Convert a numpy ndarray (HWC, uint8) to PNG bytes."""
+        import numpy as np
+
+        if not isinstance(arr, np.ndarray):
+            raise ValueError(f"Expected np.ndarray, got {type(arr)}")
```

Ruff (0.15.0)

[warning] Prefer `TypeError` exception for invalid type

(TRY004)